File: app/src/main/java/com/gplio/CustomApplication.java
package com.gplio;

import android.app.Application;

import com.squareup.leakcanary.LeakCanary;

/**
 * Created by goncalopalaio on 20/07/18.
 */

public class CustomApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            return;
        }
        LeakCanary.install(this);
    }

}

File: app/src/main/java/com/gplio/CustomApplication.kt
package com.gplio

import android.app.Application
import com.gplio.fibrewallpaper.lib.logger.AndroidLogger
import com.gplio.fibrewallpaper.lib.logger.Logger
import com.gplio.fibrewallpaper.lib.logger.Tagged
import com.gplio.fibrewallpaper.lib.logger.d

class CustomApplication : Application(), Tagged {

    override val tag: String
        get() = "CustomApplication"

    override fun onCreate() {
        super.onCreate()

        Logger.setLogger(AndroidLogger())

        d("onCreate")
    }
}
